codec/reflectcodec/type_codec.go

```diff
@@ -643,7 +670,13 @@ func (c *genericCodec) unmarshal(p *wrappers.Packer, value reflect.Value, maxSli
 		// Create a new pointer to a new value of the underlying type
 		v := reflect.New(t)
 		// Fill the value
-		if err := c.unmarshal(p, v.Elem(), c.maxSliceLen); err != nil {
+		if omitEmpty {
+			if p.UnpackBool() == true {
+				// Nil
+				return nil
+			}
```

Comment:
It seems like we typically check for `p.Err` after unpacking something.

Reply:
we should check `p.Err` as suggested above, since in the error case, the return value is false. otherwise we depend on how bool is serialized (in this case can deserialize any byte that is not 0 or 1 as if it was 0).
I'd suggest a pattern like:
```go
isNil := p.UnpackBool()
if p.Err != nil { ... }
if isNil { return nil }
```

Reply:
Also if we know the value is nil, we could avoid calling `reflect.New` here?